src/Tools/dotnet-monitor/Extensibility/ExtensionLaunchFailure.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using Microsoft.Diagnostics.Monitoring;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
+{
+    internal class ExtensionLaunchFailure : MonitoringException
```

Comment:
Let's have only one exception type (e.g. `ExtensionException : MonitoringException`) for the whole extension system. If you want to put static methods for each type of exception scenario on it, I would be fine with that e.g. `ExtensionException.ThrowNotFound(string name)` and `ExtensionException.ThrowStartFailure(string name)`.